mesonpy/_elf.py

```diff
@@ -21,9 +21,10 @@ def _patchelf(self, *args: str) -> str:
 
     @property
     def rpath(self) -> Collection[str]:
-        if not self._rpath:
-            self._rpath = self._patchelf('--print-rpath').strip().split(';')
-        return self._rpath
+        if self._rpath is None:
+            rpath = self._patchelf('--print-rpath').strip()
```

Comment:
This doesn't work portably it looks like. For example [in this CI log](https://github.com/FFY00/meson-python/actions/runs/3049095667/jobs/4914826119#step:7:340):
```
 E               subprocess.CalledProcessError: Command '['patchelf', '--print-rpath', '/tmp/pytest-of-runner/pytest-0/test_uneeded_rpath0/purelib_and_platlib-1.0.0.data/platlib/plat.{EXT_SUFFIX}']' returned non-zero exit status 1.
```